coop_shift/model/shift_mail.py

```diff
@@ -131,7 +134,14 @@ def run(self, autocommit=False):
                                   ('scheduled_date', '<=', today)])
 
         for scheduler in schedulers:
-            scheduler.execute()
-            if autocommit:
-                self.env.cr.commit()
+            try:
+                with self.env.cr.savepoint():
+                    scheduler.execute()
+            except Exception as e:
+                _logger.exception(e)
+                self.invalidate_cache()
+                self._warn_template_error(scheduler, e)
+            else:
+                if autocommit:
+                    self.env.cr.commit()
```

Comment:
Can you explain this change?
I'm worried that, in case of an exception, there is no rollback on the transaction.
And the next scheduler in the loop might commit it if it doesn't fail.

Reply:
According to this:
![image](https://user-images.githubusercontent.com/1914185/65240960-b5daf300-dae2-11e9-8833-e2e2f6a3b625.png)

Reply:
I got reference from odoo source code : https://github.com/odoo/odoo/blob/12.0/addons/event/models/event_mail.py#L155